agent/lib/internal_address_source.js

```diff
@@ -93,4 +95,98 @@ AddressSource.prototype.check_status = function (address_stats) {
     }
 };
 
+function get_configmap_name_for_address(address) {
+    return myutils.kubernetes_name(util.format('address-%s', address));
+}
+
+AddressSource.prototype.create_address = function (definition) {
+    var configmap_name = get_configmap_name_for_address(definition.address);
+    var address = {
+        apiVersion: 'enmasse.io/v1',
+        kind: 'Address',
+        metadata: {
+            name: definition.address,
+            addressSpace: this.address_space
+        },
+        spec: {
+            address: definition.address,
+            type: definition.type,
+            plan: definition.plan
+        },
+        status: {
+            isReady: 'False',
+            messages : []
+        }
+    };
+    var configmap = {
+        apiVersion: 'v1',
+        kind: 'ConfigMap',
+        metadata: {
+            name: configmap_name,
+            labels: {
+                type: 'address-config'
+            },
+            annotations: {
+                addressSpace: this.address_space,
+                cluster_id: configmap_name
```

Comment:
This should not be the name of the configmap but the name of the address (i.e. definition.address).